README.md

```diff
@@ -45,6 +45,43 @@ $stack->push(new CacheMiddleware(), 'cache');
 $client = new Client(['handler' => $stack]);
 ```
 
+# Caching strategies
+
+## Public and private 
+Theses two strategies automatically respects `Cache-Control` headers containing:
+* `no-cache`, `no-store`: response won't be cached
```

Comment:
It's more complicated than that 😄 
From the RFC:
> 5.2.1.4.  no-cache
   The "no-cache" request directive indicates that a cache MUST NOT use
   a stored response to satisfy the request without successful
   validation on the origin server.

Maybe this paragraph should explain that we try to respect the RFC.